examples/usb_i2c_spdm.rs

```diff
@@ -0,0 +1,174 @@
+//! A sample app that implements MCTP messages to be transceived from a host
+//! machine without exposed SMBus/I2C capabilities to a target endpoint using
+//! UART and I2C.
+//!
+//! The following topology is used:
+//!
+//! [HOST MACHINE] <--UART--> [USB_I2C_BRIDGE_DEVICE] <--I2C/SmBus--> [TARGET_ENDPOiNT]
+//!
+//! The host machine will issue a message to the USB_I2C_BRIDGE_DEVICE, which runs this app.
+//! The app then determines the end point target address based on the received packet header,
+//! and forwards this data to the target endpoint. As we are supporting MCTP, there are no
+//! I2C reads, only write. Thus, the endpoint must then master the bus and write a response
+//! back. This message is forwarded to the host again via UART.
+//!
+//! The host application must append a small packet header of the following format
+//! for any messages being send to this device.
+//!
+//! host_tx[0] = 0xAA // Preamble
+//! host_tx[1] = XX   // Endpoint target address
+//! host_tx[2] = YY   // MSB of 16bit data length
+//! host_tx[3] = ZZ   // LSB of 16bit data length
+//!
+//! For reception, the device (this app) first sends a packet header of the following format
+//!
+//! device_tx[0] = 0xBB // Preamble
+//! device_tx[1] = 0xFF // Unused
+//! device_tx[2] = YY   // MSB of 16bit data length
+//! device_tx[3] = ZZ   // LSB of 16bit data length
+//!
+//! Based on the data length, the host can read the next ((YY << 8) | ZZ )
+//! as data message.
+//!
+#![no_main]
+#![no_std]
+use libtock::console::Console;
+use libtock::i2c_master_slave::I2CMasterSlave;
+use libtock::leds::Leds;
+use libtock::runtime::{set_main, stack_size};
+
+set_main! {main}
+stack_size! {0x900}
+
+/// The address to which we listen for in slave/target mode.
+pub const MY_ID: u8 = 0x22;
+/// Contains packet metadata
+pub const HEADER_LEN: usize = 0x4;
+/// Max data message length
+pub const MAX_DLEN: usize = 128;
+/// Total amount of bytes we can receive in a single UART RX
+/// !! Note: When the tock kernel is built for the target, the UART_RX
+///          buffer size must be set to >= RX_BUF_LEN. Otherwise
+///          the UART driver will error out as expected due to
+///          the kernel RX buffer being smaller. Similarly,
+///          the I2CMasterSlave buffer sizes must also be
+///          >= MAX_DLEN.
```

Comment:
This should be described in the top level README. The `DEFAULT_BUF_SIZE` in the kernel needs to be set to a larger value

Reply:
I added a new README for the examples, which we can use to add application specific info. Thoughts?

Reply:
I meant just in the top level comments

Reply:
ah got it, fixed!